packages/puppeteer/test/axePuppeteer.test.ts

```diff
@@ -0,0 +1,711 @@
+
+        const axePip = new AxePuppeteer(page, axeSource).include('.include');
+
+        await expectAsyncToNotThrow(() => axePip.analyze());
+      });
+    });
+
+    describe('with only exclude', () => {
+      it('adds .exclude to context', async () => {
+        const axeSource = `
+          window.axe = {
+            configure () {},
+            run (context, options, config) {
+              if (context === document) {
+                return Promise.reject(new Error('Invalid context'))
+              }
+              if (context.include) {
+                return Promise.reject(new Error('Invalid include context'))
+              }
+
+              if (context.exclude[0] !== '.exclude') {
+                return Promise.reject(new Error('Invalid exclude context'))
+              }
+
+              return Promise.resolve({})
+            }
+          }
+        `;
+
+        await page.goto(`${addr}/context.html`);
+
+        const axePip = new AxePuppeteer(page, axeSource).exclude('.exclude');
+
+        await expectAsyncToNotThrow(() => axePip.analyze());
+      });
+    });
+
+    it('defaults to document', async () => {
+      const axeSource = `
+          window.axe = {
+            configure () {},
+            run (context, options, config) {
+              if (context !== document) {
+                return Promise.reject(new Error('Invalid context'))
+              }
+
+              return Promise.resolve({})
+            }
+          }
+        `;
+
+      await page.goto(`${addr}/context.html`);
+
+      const axePip = new AxePuppeteer(page, axeSource);
+
+      await expectAsyncToNotThrow(() => axePip.analyze());
+    });
+
+    describe('.disableFrame()', () => {
+      it('disables the given rule(s)', async () => {
+        await page.goto(`${addr}/external/nested-iframes.html`);
+        const results = await new AxePuppeteer(page)
+          // Ignore all frames
+          .disableFrame('#ifr-foo, #ifr-bar')
+          .disableFrame('#ifr-baz')
+          .analyze();
+
+        const labelResult = results.violations.find(
+          (r: Axe.Result) => r.id === 'label'
+        );
+        expect(labelResult).to.be.undefined;
+      });
+    });
+  });
+
+  describe('.configure()', () => {
+    it('accepts custom configuration', async () => {
+      const config: Axe.Spec = {
+        checks: [
+          {
+            evaluate: (): false => false,
+            id: 'foo'
+          }
+        ],
+        rules: [
+          {
+            all: [],
+            any: ['foo'],
+            id: 'foo',
+            none: [],
+            selector: 'html',
+            tags: ['wcag2aa']
+          }
+        ]
+      };
+
+      await page.goto(`${addr}/external/index.html`);
+
+      // HACK: work around axe-core (incorrectly) requiring this to be
+      // a function (see https://github.com/dequelabs/axe-core/issues/974).
+      (config.checks as any)[0].evaluate = 'function () { return false }';
```

Comment:
I believe this should no longer be required as evaluate can be a string [in the typings](https://github.com/dequelabs/axe-core/blob/develop/axe.d.ts#L213)

```suggestion
      config.checks[0].evaluate = 'function () { return false }';
```

Reply:
Copied this from the previous implementation, but yeah it can be removed.